test/model/message_list_test.dart

```diff
@@ -2244,8 +2813,16 @@ void checkInvariants(MessageListView model) {
   check(model).middleItem
     ..isGreaterOrEqual(0)
     ..isLessOrEqual(model.items.length);
-  if (model.middleItem == model.items.length) {
-    check(model.middleMessage).equals(model.messages.length);
+  if (model.middleMessage == model.messages.length) {
+    if (model.outboxMessages.isEmpty) {
+      // the bottom slice of `model.messages` is empty
+      check(model.items).length.equals(model.middleItem);
+    } else {
+      check(model.items.sublist(model.middleItem))
+        ..first.isA<MessageListOutboxMessageItem>()
+            .message.identicalTo(model.outboxMessages.first)
+        ..every((item) => item.not((it) => it.isA<MessageListMessageItem>()));
```

Comment:
This "is not MessageListMessageItem" check is redundant with the "is MessageListOutboxMessageItem" check just above, given the overall structure of `items`: all items for outbox-messages come after all items for normal messages. And that latter structure was verified in the big `for (int j = 0; j < allMessages.length; j++)` loop above.

So this `every` can be dropped. And then the `.sublist` plus `.first` can be simplified down.